### ISSUE TITLE
potential overfitting

### ISSUE THREAD

**Issue by sky1ove** [bytedance/Protenix#123]
Dear Protenix team,

I recently tried running kras mutant inhibitor docking using protenix. Take pdb 7T47 for example, if taking protein sequence of 7T47 "MGHHHHHHSSGRENLYFQGMTEYKLVVVGADGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVIDGETSLLDILDTAGQEEYSAMRDQYMRTGEGFLLVFAINNTKSFEDIHHYREQIKRVKDSEDVPMVLVGNKSDLPSRTVDTKQAQDLARSYGIPFIETSAKTRQGVDDAFYTLVREIRKHKEK" as protein sequence input, and take 6IC (CCD) as ligand, it can almost predict the position correctly. 

However, if using the kras mutant sequence "MTEYKLVVVGADGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVIDGETCLLDILDTAGQEEYSAMRDQYMRTGEGFLCVFAINNTKSFEDIHHYREQIKRVKDSEDVPMVLVGNKCDLPSRTVDTKQAQDLARSYGIPFIETSAKTRQRVEDAFYTLVREIRQYRLKKISKEEKTPGCVKIKKCIIM" (compared with the first sequence, it doesn't contain the His tag and two single mutations for stability of crystal), it can't predict the position of the ligand (totally out of the original ligand box)

<img width="674" alt="Image" src="https://github.com/user-attachments/assets/d905b572-513a-456f-9937-4814f8dccf8c" />

I tried both local and the server and got the similar results.

I also tested AF3, which can predict this correctly.

**Comment by zhangyuxuann** [comment]
@sky1ove Thanks for your feedback. 7T47 is in the test set, not in the training set. We will check the case you provided.

**Comment by sky1ove** [comment]
> [@sky1ove](https://github.com/sky1ove) Thanks for your feedback. 7T47 is in the test set, not in the training set. We will check the case you provided.

Hi, thanks for the update. It is correct in the new version:

![Image](https://github.com/user-attachments/assets/6033a877-91a5-4328-9eb1-3a455c1e7d4c)

**Comment by xinshi-chen** [comment]
> > [@sky1ove](https://github.com/sky1ove) Thanks for your feedback. 7T47 is in the test set, not in the training set. We will check the case you provided.
> 
> Hi, thanks for the update. It is correct in the latest version.

Thanks for your follow-up! Glad to hear it's working correctly in the latest version. We'll go ahead and close this issue — feel free to reach out again if you encounter anything else!
